### PR TITLE
Facepile Not Announcing Position in Set Correctly

### LINKED ISSUE #7534: Facepile A11y Pass
Tracking all of the issues brought up with the 1/4 accessibility pass on Facepile control 
Details on individual bugs can be found on [SharePoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7Bfd43fcc5-515d-47d8-946d-0d4862a56278%7D&action=default&uid=%7BFD43FCC5-515D-47D8-946D-0D4862A56278%7D&ListItemId=1647&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble) 
  
 - [ ] Bug1: [Accessibility][Programmatic access]: Certain ARIA roles must be contained by particular parents. 
 - [ ] Bug2: [Accessibility][Keyboard navigation]: User is not able to access all facepile controls using keyboard.  
 - [ ] Bug3: [Accessibility][Usable]: In scan mode narrator doesn't announce role and state of the 'Overflow Button Type',  'Persona Size' list box.  
 - [ ] Bug4: [Accessibility][Usable]: Narrator doesn't announce the slider value along with the control name while changing the 'Number of persona' slider using arrow keys. 
 - [ ] Bug5: [Accessibility][Functional]: No option is given to remove the facepile control present under 'Facepile with face adding functionality' section. 
 - [ ] Bug6: [Accessibility][Screen reader]: Narrator announces same position number for all facepile controls.  
 - [ ] Bug7: [Accessibility][Usable]: Narrator scan mode focus is not visible on facepile controls as screen gets scroll down once user navigates on them.   
 - [ ] Bug8: [Accessibility][Programmatic access]: Screen reader doesn't give any feedback while adding facepiles using 'Add friend' button. 
 
 
 Component: Facepile 
@micahgodbolt

### PR DESCRIPTION
**Pull request by tabrumle**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixing #7534 Bug 6
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

FacepileButtons were not correctly labelling their position in the Facepile. Both the li item containing the FacepileButton and the FacepileButton itself were marked with the "option" role. Change the li role to "Presentation". Tested the change out and was able to get the expected behavior where each FacepileButton would announce the persona name and its position in the Facepile.

#### Focus areas to test

Accessibility with narrator.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9823)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.base.tsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ export class FacepileBase extends BaseComponent<IFacepileProps, {}> {
         ? onRenderPersona(persona, this._getPersonaControl)
         : onRenderPersonaCoin(persona, this._getPersonaCoinControl);
       return (
-        <li role="option" key={`${singlePersona ? 'persona' : 'personaCoin'}-${index}`} className={this._classNames.member}>
+        <li role="presentation" key={`${singlePersona ? 'persona' : 'personaCoin'}-${index}`} className={this._classNames.member}>
```
Comment (Markionium) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.base.tsx]: The snapshots need to be updated. You can start the tests in watch mode using `npm run start-test` (office-ui-fabric-react pacakge) and press `u` to update the snapshots.

You might need to update some other snapshots too for examples. But i recommend you to run a full `npm run build` in the root, so you do not need to wait for CI each time :D

File: change/office-ui-fabric-react-2019-07-16-09-47-56-tabrumle-FixFacepileAriaPosInSet.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "comment": "Fix Facepile Position in set for narrator",
```
Comment (Vitalius1) [inline change/office-ui-fabric-react-2019-07-16-09-47-56-tabrumle-FixFacepileAriaPosInSet.json]:
```suggestion
  "comment": "Facepile: fix screenreader readout of PersonaCoin position in the set.”,
```

**Comment by Vitalius1** [comment]
@tabrumle is there a reason you closed the PR?

**Comment by tabrumle** [comment]
Unfamiliar with GitHub PR's. Is closing not the same as completing the PR?

**Comment by Vitalius1** [comment]
Will wait till the CI build will pass with green light and then merge it in. I will complete the merge as you don’t have permissions to do it.

**Comment by Vitalius1** [comment]
@msft-github-bot merge this PR in 1 hour

**Comment by tabrumle** [comment]
Looking at  issue #[7695](https://github.com/OfficeDev/office-ui-fabric-react/issues/7667) it looks like Keros was picking up an error because we weren't using "listbox" originally and then there was a subsequent issue #[8048](https://github.com/OfficeDev/office-ui-fabric-react/issues/8048) that stemmed from the <li> not having the "option" role, so we probably do need to keep the same roles. Staying with the same role for the li means that both the li and its child element will be announced during caps+arrows narrator navigation. I'm pushing a change that will fix this by removing the "option" role from the button and adding the aria-hidden=true attribute. This will prevent announcement during caps+arrow navigation but will still still announce the button when focused.

**Comment by KevinTCoughlin** [comment]
@tabrumle thanks for the follow-up. Your rationale seems sounds to me.

Can you run `npm run update-a11y` from the root of the repo to update our accessibility snapshot output and commit the result?

This will unblock CI and let us see the changes on the deploy site.

**Comment by Vitalius1** [comment]
@tabrumle sorry for the friction 😒  I think you would have to first build up to `a11y-tests` project and then update the snapshots like Kevin mentioned because I still see the CI failing on the accessibility test snapshot. Thanks!

**Comment by tabrumle** [comment]
No worries, and thanks for the feedback!

**Pull request merged**